EasyPost/PaymentMethod.cs

```diff
@@ -1,38 +1,100 @@
-using System;
 using System.Threading.Tasks;
 using Newtonsoft.Json;
-using RestSharp;
 
 namespace EasyPost
 {
+    /// <summary>
+    ///     Represents a summary of the primary and secondary payment methods on the user's account.
+    /// </summary>
     public class PaymentMethod : Resource
     {
+        /// <summary>
+        ///     Payment method priority
+        /// </summary>
+        public enum Priority
+        {
+            Primary,
+            Secondary
+        }
+
         [JsonProperty("id")]
         public string id { get; set; }
         [JsonProperty("object")]
         public string Object { get; set; }
         [JsonProperty("primary_payment_method")]
-        public CreditCard primary_payment_method { get; set; }
+        public PaymentMethodObject primary_payment_method { get; set; }
         [JsonProperty("secondary_payment_method")]
-        public CreditCard secondary_payment_method { get; set; }
+        public PaymentMethodObject secondary_payment_method { get; set; }
+    }
+
+    /// <summary>
+    ///     Represents a credit card or a bank account.
+    /// </summary>
+    public class PaymentMethodObject : Resource
+    {
+        public enum PaymentMethodType
+        {
+            CreditCard,
+            BankAccount
+        }
+
+        [JsonProperty("bank_name")]
+        public string bank_name { get; set; }
+        [JsonProperty("brand")]
+        public string brand { get; set; }
+        [JsonProperty("country")]
+        public string country { get; set; }
+        [JsonProperty("disabled_at")]
+        public string disabled_at { get; set; }
+        [JsonProperty("exp_month")]
+        public string exp_month { get; set; }
+        [JsonProperty("exp_year")]
+        public string exp_year { get; set; }
+        [JsonProperty("id")]
+        public string id { get; set; }
+        [JsonProperty("last4")]
+        public string last4 { get; set; }
+        [JsonProperty("name")]
+        public string name { get; set; }
+        [JsonProperty("object")]
+        public string Object { get; set; }
 
         /// <summary>
-        ///     List all payment methods for this account.
+        ///     Get what type of payment method this is (credit card, bank account, etc.)
         /// </summary>
-        /// <returns>An EasyPost.PaymentMethod summary object.</returns>
-        /// <exception cref="Exception"></exception>
-        public static async Task<PaymentMethod> All()
+        public PaymentMethodType? Type
         {
-            Request request = new Request("payment_methods", Method.Get);
+            get
+            {
+                if (id == null)
+                {
+                    return null;
+                }
```

Comment:
This can be removed

Reply:
This safeguards in case somehow the user has created a PaymentMethodObject object without initializing it.

Reply:
I mean you have a return statement that return `null` down below, this if-statement is redundant

Reply:
I think if I tried to call `StartsWith` on a `null` value, it would throw an error, so this prevents that.